src/mantine-nprogress/src/NProgress.tsx

```diff
@@ -0,0 +1,126 @@
+import { MantineTransition, OptionalPortal, Progress, Transition } from '@mantine/core';
+import { useDebouncedValue, useDidUpdate, useInterval, useReducedMotion } from '@mantine/hooks';
+import { getDefaultZIndex, MantineColor } from '@mantine/styles';
+import React, { useState } from 'react';
+import { useNProgressEvents } from './events';
+
+export interface NProgressProps {
+  /** The default progress */
+  defaultProgress?: number;
+
+  /** The color of the progressbar */
+  color?: MantineColor;
+
+  /** The height of the progressbar */
+  size?: number;
+
+  /** Called when the progress is 100% */
+  onFinish?: () => void;
+
+  /** Interval time in ms */
+  intervalTime?: number;
```

Comment:
Almost all props that involve timeouts are named poorly – it is not clear what they do, please review them and add better names and descriptions.